src/test/java/nl/knaw/dans/dvcli/action/SingleOrTest.java

```diff
@@ -0,0 +1,188 @@
+            .containsExactlyInAnyOrderElementsOf(List.of(
+                new Pair<>("a", datasetA),
+                new Pair<>("1", dataset1),
+                new Pair<>("blabla", datasetBlabla)
+            ));
+
+        verify(dataverseClient, times(1)).dataset("a");
+        verify(dataverseClient, times(1)).dataset(1);
+        verify(dataverseClient, times(1)).dataset("blabla");
+        verifyNoMoreInteractions(dataverseClient);
+    }
+
+    @Test
+    public void getDatasets_should_parse_stdin_and_return_empty_lines() throws IOException {
+
+        var dataverseClient = mock(DataverseClient.class);
+
+        var datasetA = mock(DatasetApi.class);
+        Mockito.when(dataverseClient.dataset("A")).thenReturn(datasetA);
+        var datasetBlank = mock(DatasetApi.class);
+        Mockito.when(dataverseClient.dataset("")).thenReturn(datasetBlank);
+        var datasetBlabla = mock(DatasetApi.class);
+        Mockito.when(dataverseClient.dataset("rabarbera")).thenReturn(datasetBlabla);
+        var dataset1 = mock(DatasetApi.class);
+        Mockito.when(dataverseClient.dataset("B")).thenReturn(dataset1);
+
+        System.setIn(new ByteArrayInputStream("""
+            A
+                        
+            B rabarbera
+                        
+            """.getBytes()));
+
+        var datasets = new SingleDatasetOrDatasetsFile("-", dataverseClient)
+            .getDatasets();
+        Assertions.assertThat(datasets.toList())
+            .containsExactlyInAnyOrderElementsOf(List.of(
+                new Pair<>("A", datasetA),
+                new Pair<>("B", dataset1),
+                new Pair<>("rabarbera", datasetBlabla),
+                new Pair<>("", datasetBlank),
+                new Pair<>("", datasetBlank)
+            ));
+
+        verify(dataverseClient, times(1)).dataset("A");
+        verify(dataverseClient, times(1)).dataset("B");
+        verify(dataverseClient, times(1)).dataset("rabarbera");
+        verify(dataverseClient, times(2)).dataset("");
+        verifyNoMoreInteractions(dataverseClient);
+    }
+
+    @Test
+    public void getDatasets_should_read_until_Exception() throws IOException {
+
+        var dataverseClient = mock(DataverseClient.class);
+
+        Mockito.when(dataverseClient.dataset("A"))
+            .thenReturn(mock(DatasetApi.class));
+        Mockito.when(dataverseClient.dataset("whoops"))
+            .thenThrow(new RuntimeException("test"));
+
+        System.setIn(new ByteArrayInputStream("""
+            A
+            whoops
+            B""".getBytes()));
+
+        var datasets = new SingleDatasetOrDatasetsFile("-", dataverseClient)
+            .getDatasets();
+        assertThatThrownBy(datasets::toList)
+            .isInstanceOf(RuntimeException.class)
+            .hasMessage("test");
+
+        verify(dataverseClient, times(1)).dataset("A");
+        verify(dataverseClient, times(1)).dataset("whoops");
+        verify(dataverseClient, times(0)).dataset("B");
```

Comment:
This behavior looks like the fail-fast option of the python batch processor. Default behavior continued.

Reply:
I see/remember now that the service is not accessed at this stage. So my remark makes no sense, an exception is even very unlikely.